juan-silva/tarea-5/puzzle.js

```diff
@@ -0,0 +1,94 @@
+//Initial matrix
+let matrixPuzzle=[[2,3,10,11],[14,1,13,15],[5,4,8,7],[6,12,9,""]];
+//explore the positions around the empty space
+function findMoves(){
+    let matrix = getMatrix();
+    let posZero = findZeroPos(matrix);
+    let moves=[];
+    //Each if has directions(up,down,left,right) where the Zero can move
+    posZero[0]<=3 && posZero[0]>0?moves.push([posZero[0]-1,posZero[1],"up"]):null;
+    posZero[0]<3 && posZero[0]>=0?moves.push([posZero[0]+1,posZero[1],"down"]):null;
+    posZero[1]<=3 && posZero[1]>0?moves.push([posZero[0],posZero[1]-1,"left"]):null;  
+    posZero[1]<3 && posZero[0]>=0?moves.push([posZero[0],posZero[1]+1,"right"]):null;
+    return moves;
+}
+//find the empty space in the puzzle
+function findZeroPos(){
+    let matrix = getMatrix();
+    let positionZero=[];
+    for(var i=0; i<matrix.length; i++) {

```

Comment:
@jsilva97 , use modern approach like forEach, map or reduce, also remember name properly variables i and j